math_utils/number_theory.py

```diff
@@ -0,0 +1,431 @@
+"""
+Number theory utilities module.
+
+This module provides mathematical functions for prime numbers, Fibonacci sequences,
+perfect numbers, and other number theory operations with proper input validation
+and error handling.
+"""
+
+import math
+from typing import List, Iterator
+
+
+def is_prime(n: int) -> bool:
+    """
+    Check if a number is prime.
+
+    Args:
+        n: Integer to check for primality
+
+    Returns:
+        True if n is prime, False otherwise
+
+    Raises:
+        TypeError: If n is not an integer
+        ValueError: If n is less than 2
+
+    Examples:
```

Comment:
the examples in the function comments in general are taking up too much space. let's have their return and call on one single line.

if we can make the function comments more compact in general that would be good. for example, no new lines in between the different section; the `header:` and indentation provide enough of a break